pulser-core/pulser/backend/abc.py

```diff
@@ -26,20 +27,46 @@
 class Backend(ABC):
     """The backend abstract base class."""
 
-    def __init__(self, sequence: Sequence) -> None:
+    def __init__(self, sequence: Sequence, mimic_qpu: bool = False) -> None:
         """Starts a new backend instance."""
-        self.validate_sequence(sequence)
+        self.validate_sequence(sequence, mimic_qpu=mimic_qpu)
         self._sequence = sequence
+        self._mimic_qpu = bool(mimic_qpu)
 
     @abstractmethod
     def run(self) -> Results | typing.Sequence[Results]:
         """Executes the sequence on the backend."""
         pass
 
-    def validate_sequence(self, sequence: Sequence) -> None:
+    @staticmethod
+    def validate_sequence(sequence: Sequence, mimic_qpu: bool = False) -> None:
         """Validates a sequence prior to submission."""
         if not isinstance(sequence, Sequence):
             raise TypeError(
                 "'sequence' should be a `Sequence` instance"
                 f", not {type(sequence)}."
             )
+        if not mimic_qpu:
+            return
```

Comment:
nitpick: This method signature looks a bit odd to me. 

- The name `mimic_qpu` feels slightly out of place in the scope of this method. It feels appropriate when instantiating a backend object - which will mimic a qpu - but this function scope is only to validate a sequence - in that context `mimic_qpu` doesn't make full sense to me. If we kept it more functional by renaming it `skip_device_specs_validation` (very bad name but you get the idea) it would make that function clearer I think? The caller would know whether to skip these validation steps depending on whether it attempts to mimic a QPU or not. An alternative would be to have a method for the type checking of the sequence and another one for the validation against the device specs and it would be the caller's responsibility to know when to apply each of them - this may cause some duplication you want to avoid though.  Maybe I'm not seeing this through the right lense though.
- Is there any reason why this is a method of the Backend class and not a method of the sequence class?

Reply:
Thanks a lot for the review @MatthieuMoreau0 :pray: 
I think perhaps the name of the method is the confusing part... `validate_sequence()` is not validating the sequence against the device specs (that happens while the sequence is being built), but rather validating it for submission on the backend. When you submit a sequence to a QPUBackend, you have extra conditions you have to meet, like the device having to be a `Device` instance and not `VirtualDevice`, or the register having to come from a layout. These conditions cannot be enforced at the `Sequence` level because the sequence does not know which backend it will be executed in. Of course, we could add a `Sequence.validate_for_qpu_submission()` method, but what would be the benefit in that?

Reply:
On that note, the `mimic_qpu` flag is just telling the method to mimic a QPU validation, so it doesn't seem so weird to me...

Reply:
Oh okay, I see my bad. I'll resolve this comment. As per one of my comments above I think this is a bit convoluted but I simply lack familiarity with Pulser